test-tosam.sh

```diff
@@ -4,6 +4,8 @@ success=1
 for f in test-data/*.mas; do
     if ! cmp -s <(./tosam $f) `basename $f .mas`.sam; then
```

Comment:
You haven't fixed the [error mentioned here](https://github.com/orgs/birc-gsa-2022/discussions/1#discussioncomment-3507434). That means that the test, and not necessarily your code, is failing.